From 1d087dfcea16cde51b103bfdc7b1303b14eb3c11 Mon Sep 17 00:00:00 2001
From: Jeremy Drake <github@jdrake.com>
Date: Thu, 11 Jan 2024 17:56:27 -0800
Subject: [PATCH] rust: build gnullvm targets for clang

---
 mingw-w64-rust/PKGBUILD | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/mingw-w64-rust/PKGBUILD b/mingw-w64-rust/PKGBUILD
index e345a3b97..0486d5983 100644
--- a/mingw-w64-rust/PKGBUILD
+++ b/mingw-w64-rust/PKGBUILD
@@ -4,7 +4,7 @@
 # Contributor: Philippe Renon <philippe_renon@yahoo.fr>
 
 _bootstrapping=yes
-if [[ $MINGW_PACKAGE_PREFIX == *-clang-aarch64 ]]; then
+if [[ $MINGW_PACKAGE_PREFIX == *-clang-* ]]; then
   _bootstrapping=no
 fi
 
@@ -108,10 +108,6 @@ prepare() {
     0005-win32-config.patch \
     0008-disable-self-contained.patch
 
-  if [[ $MINGW_PACKAGE_PREFIX == *-clang-i686 || $MINGW_PACKAGE_PREFIX == *-clang-x86_64 ]]; then
-    apply_patch_with_msg \
-      0007-clang-subsystem.patch
-  fi
   if [[ $MINGW_PACKAGE_PREFIX == *-clang-* ]]; then
     apply_patch_with_msg \
       0011-disable-uac-for-installer.patch
@@ -151,7 +147,7 @@ build() {
       "build/cache/${_stage0date}/"
   fi
 
-  if [[ $MINGW_PACKAGE_PREFIX == *-clang-aarch64 ]]; then
+  if [[ $MINGW_PACKAGE_PREFIX == *-clang-* ]]; then
     # We have to do the following because rust doesn't count x86_64-w64-mingw32 as a target triple
     OSTYPE="$CARCH-pc-windows-gnullvm"
 
-- 
2.45.1.windows.1

